packages/x-components/src/x-modules/tagging/store/actions/track-tagging.action.ts

```diff
@@ -0,0 +1,42 @@
+import { TaggingInfo } from '../../../../../../search-types';
+import { XPlugin } from '../../../../plugins/x-plugin';
+import { DefaultSessionService } from '../../service';
+import { TaggingXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link TaggingActions.track}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ * @param taggingInfo - The information of the event to track.
+ *
+ * @public
+ */
+export const track: TaggingXStoreModule['actions']['track'] = ({ state }, taggingInfo) => {
+  const { consent } = state;
+  const tagging = Array.isArray(taggingInfo) ? taggingInfo : [taggingInfo];
+  const sessionId = getSessionId(consent);
+
+  tagging.forEach(({ url, params }: TaggingInfo) => {
+    XPlugin.adapter.track({
+      url,
+      params: {
+        ...params,
+        session: sessionId
```

Comment:
Our `BeaconTrackingRequestor` isn't as smart to remove `undefined` values, so this is sending the string `"undefined"` to the API as the `session` parameter when you have `consent: false`.

Reply:
You're right, I saw it yesterday but I forgot about it sorry.